src/Native/System.Native/pal_io.cpp

```diff
@@ -974,21 +975,48 @@ extern "C" int32_t SystemNative_CopyFile(intptr_t sourceFd, intptr_t destination
     struct stat_ sourceStat;
     bool copied = false;
 #if HAVE_SENDFILE
+    // If sendfile is available (Linux), try to use it, as the whole copy
+    // can be performed in the kernel, without lots of unnecessary copying.
     while (CheckInterrupted(ret = fstat_(inFd, &sourceStat)));
     if (ret != 0)
     {
         return -1;
     }
 
-    // If sendfile is available (Linux), try to use it, as the whole copy 
-    // can be performed in the kernel, without lots of unnecessary copying.
-    if (sendfile(outFd, inFd, 0, UnsignedCast(sourceStat.st_size)) >= 0)
+
+#if defined(BIT32) && defined(HAVE_STAT64)
+    // On 32-bit the last argument of `sendfile' will be `size_t' a 32-bit integer
+    // while the `st_size' field of the stat structure will be off64_t.
+    // We tweak the type of `size' to have code that works regardless of this.
+    uint64_t size;
+#else
+    size_t size;
+#endif
+    size = UnsignedCast(sourceStat.st_size);
```

Comment:
I'm not sure how others feel about this (cc: @nguerrera, @sokket, @eerhardt), but what about just using ```auto``` here, e.g.
```C++
auto size = UnsignedCast(sourceState.st_size);
```
rather than having these ifdefs?